lib/ipam/ipam.go

```diff
@@ -1391,18 +1518,31 @@ func (c ipamClient) GetAssignmentAttributes(ctx context.Context, addr net.IP) (m
 // GetIPAMConfig returns the global IPAM configuration.  If no IPAM configuration
 // has been set, returns a default configuration with StrictAffinity disabled
 // and AutoAllocateBlocks enabled.
-func (c ipamClient) GetIPAMConfig(ctx context.Context) (*IPAMConfig, error) {
+func (c ipamClient) GetIPAMConfig(ctx context.Context) (config *IPAMConfig, err error) {
 	obj, err := c.client.Get(ctx, model.IPAMConfigKey{}, "")
 	if err != nil {
 		if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
 			// IPAMConfig has not been explicitly set.  Return
 			// a default IPAM configuration.
-			return &IPAMConfig{AutoAllocateBlocks: true, StrictAffinity: false}, nil
+			config = &IPAMConfig{AutoAllocateBlocks: true, StrictAffinity: false}
+			err = nil
+		} else {
+			log.Errorf("Error getting IPAMConfig: %v", err)
+			return nil, err
 		}
-		log.Errorf("Error getting IPAMConfig: %v", err)
-		return nil, err
+	} else {
+		config = c.convertBackendToIPAMConfig(obj.Value.(*model.IPAMConfig))
+	}
+	if detectOS(ctx) == "windows" {
+		// When a Windows node owns a block, it creates a local /26 subnet object and as far as we know, it can't
+		// do a longest-prefix-match between the subnet CIDR and a remote /32.  This means that we can't allow
+		// remote hosts to borrow IPs from a Windows-owned block; and Windows hosts can't borrow IPs either.
+		// Force Windows nodes to always behave as if StrictAffinity is enabled.  This means that they won't borrow
+		// and they'll mark any blocks that they allocate as Strict so others can't borrow from them.
```

Comment:
Hm, this is a problem - the Linux code doesn't look at the block.StrictAffinity field any more, and instead only looks at the global configuration. We changed this in a recent release to enable the strict affinity code to behave more naturally.

I think that means that, instead of forcing this to be "true" here, we need to return an error if the global strict affinity configuration is set to "false" and we detect that we're running on a Windows node. That will prevent Windows nodes from launching pods at all so long as strict affinity is not enabled.

WDYT?

Reply:
The problem is some user would like to join windows nodes to an exiting cluster which has global strict affinity set to `false`. This would not going to work for them.

Reply:
But what can we do about that?

I think the only thing we can reasonably do is to reject creation of pods on Windows nodes if the global strict affiinity is false. 

The problem is, there is no longer any concept of a per-node strictaffinity option. It was removed.